twister blackbox test test_inline_logs fails on python 3.10 in CI

**Describe the bug**

See https://github.com/zephyrproject-rtos/zephyr/actions/runs/14092198977/job/39471377995?pr=87609

**To Reproduce**

Cant reproduce locally with the same python version

**Expected behavior**

test should pass.

**Impact**

Prevents changes from going in....

**Logs and console output**

```
        split_build_log = build_log.split('\n')
        for r in split_build_log:
>           assert r in inline_twister_log
E           AssertionError: assert 'Building C object zephyr/arch/arch/x86/core/CMakeFiles/arch__x86__core.dir/spec_ctrl.c.obj' in '2025-03-26 22:20:54,974 - twister - INFO - Using Ninja..\n2025-03-26 22:20:55,233 - twister - INFO - Zephyr version: ...-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+\n2025-03-26 22:21:14,845 - twister - INFO - Run completed\n'



```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used

**Additional context**
<!--
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...
-->

--- a/scripts/tests/twister_blackbox/test_output.py
+++ b/scripts/tests/twister_blackbox/test_output.py
@@ -82,7 +82,7 @@ class TestOutput:
             assert all([(tc_name.count('.') == 1) for _, _, tc_name in filtered_j])
 
 
-    def xtest_inline_logs(self, out_path):
+    def test_inline_logs(self, out_path):
         test_platforms = ['qemu_x86', 'intel_adl_crb']
         path = os.path.join(TEST_DATA, 'tests', 'always_build_error', 'dummy')
         args = ['--outdir', out_path, '-T', path] + \
@@ -132,7 +132,11 @@ class TestOutput:
             r'-- Configuring done \([0-9.]+s\)',
             r'-- Generating done \([0-9.]+s\)',
             # Cache location may vary between CI runs
-            r'^.*-- Cache files will be written to:.*$'
+            r'^.*-- Cache files will be written to:.*$',
+            # List of built C object may differ between runs.
+            # See: Issue #87769.
+            # Probable culprits: the cache mechanism, build error
+            r'^Building C object .*$'
         ]
         for pattern in removal_patterns:
             c_pattern = re.compile(pattern, flags=re.MULTILINE)

